tests/test-maximize.lua

```diff
@@ -99,6 +100,47 @@ local steps = {
             assert(new_geo[k] == v)
         end
 
+        c.floating  = true
+
+        awful.placement.centered(c)
+        original_geo = c:geometry()
+
+        c.maximized = true
+
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        local new_geo = c:geometry()
+        local sgeo    = c.screen.workarea
+
+        assert(c.maximized)
+        assert(c.floating)
+        assert(new_geo.x==sgeo.x)
+        assert(new_geo.y==sgeo.y)
+
+        c.maximized = false
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if c.maximized then
+            return
+        end
```

Comment:
Is this `if` necessary? Setting `c.maximized` should be instantaneous (so this could even be done in the previous step)